modules/update_helper_checklist/src/UpdateChecklist.php

```diff
@@ -18,6 +17,11 @@
  */
 class UpdateChecklist {
 
+  /**
+   * The configuration key for saved progress.
+   */
+  const PROGRESS_CONFIG_KEY = 'progress';
```

Comment:
Doing this doesn't look right - I think we should be using state for storing progress and not configuration ala https://www.drupal.org/project/checklistapi/issues/2944103

There's new methods setSavedProgress() and getSavedProgress()